source/Halibut/Transport/Protocol/MessageSerializer.cs

```diff
@@ -88,6 +104,9 @@ T ReadCompressedMessageRewindable<T>(Stream stream, IRewindableBuffer rewindable
                     {
                         rewindable.CancelBuffer();
                     }
+
+                    observer.MessageRead(decompressedObservableStream.BytesRead);
```

Comment:
Could this be pushed into the observable stream, to avoid needing to remember to call it?

Reply:
I might leave it where it is so that if we enhance observability beyond this stream, we can do so easily.